SystemMicrocontroller_Sensors/Core/Src/lcd20x4.c

```diff
@@ -0,0 +1,331 @@
+/*
+ * lcd20x4.c
+ *
+ *  Created on: May 15, 2021
+ *      Author: jdrqu
+ */
+
+#include "lcd20x4.h"
+
+const uint8_t ROW_20[] = {0x00, 0x40, 0x14, 0x54}; //address of the first position of each row for a 20x4 LCD
+
+
+void LCD_Init(LCD_HandleTypeDef * lcd){
+
+	LCD_PinInit(lcd);
+	DELAY(5);
+	if(lcd->Mode == LCD_MODE_4BITS)
+	{
+		HAL_GPIO_WritePin(lcd->RS_port, lcd->RS_pin, LCD_RS_COMMAND);		// Write to command register
+		DELAY(50); //wait > 40 ms
+		LCD_Write(lcd, 0x3);
+		DELAY(5); //wait >4.1 ms
+		LCD_Write(lcd, 0x3);
+		DELAY(1); //wait >100 us
+		LCD_Write(lcd, 0x3);
+		DELAY(1);
+		LCD_Write(lcd, 0x2);
+		DELAY(1);
+		LCD_WriteCommand(lcd, COMMAND_FUNCTION_SET | OPT_N); //2 lines, 4 bit data length
+		LCD_WriteCommand(lcd, COMMAND_DISPLAY_ON_OFF_CONTROL);
+		LCD_WriteCommand(lcd, COMMAND_CLEAR_DISPLAY);
+		LCD_WriteCommand(lcd, COMMAND_ENTRY_MODE_SET | OPT_INC); // Increment cursor
+		LCD_WriteCommand(lcd, COMMAND_DISPLAY_ON_OFF_CONTROL | OPT_D | OPT_C | OPT_B); //turn on display, cursor and cursor blink
+	}
+	else
+	{
+		DELAY(45); //wait > 40 ms
+		LCD_WriteCommand(lcd, 0x30);
+		DELAY(5); //wait >4.1 ms
+		LCD_WriteCommand(lcd, 0x30);
+		DELAY(1); //wait >100 us
+		LCD_WriteCommand(lcd, 0x30);
+		DELAY(1);
+		LCD_WriteCommand(lcd, COMMAND_FUNCTION_SET | OPT_DL | OPT_N); //2 lines, 8 bit data length
+		LCD_WriteCommand(lcd, COMMAND_DISPLAY_ON_OFF_CONTROL | OPT_D | OPT_C | OPT_B); //turn on display, cursos and cursor blink
+		LCD_WriteCommand(lcd, COMMAND_CLEAR_DISPLAY);
+		LCD_WriteCommand(lcd, COMMAND_ENTRY_MODE_SET | OPT_INC); // Increment cursor
+	}
+}
+
+void LCD_PinInit(LCD_HandleTypeDef * lcd) {
```

Comment:
Detailed design is not a requirement yet, but I think function header should be add with a brief of what the function does. Its valid inputs, and its output.